src/app/content/components/Sidebar.tsx

```diff
@@ -1,59 +1,106 @@
-// tslint:disable:variable-name
 import { HTMLElement } from '@openstax/types/lib.dom';
 import React, { Component, ComponentType } from 'react';
-import { FormattedMessage } from 'react-intl';
 import { connect } from 'react-redux';
 import styled, { css } from 'styled-components';
+import MobileScrollLock from '../../components/MobileScrollLock';
+import { navDesktopHeight, navMobileHeight } from '../../components/NavBar';
 import theme from '../../theme';
 import { AppState } from '../../types';
 import { isArchiveTree } from '../guards';
 import * as selectors from '../selectors';
-import { ArchiveTree, Book, Page } from '../types';
-import { scrollTocSectionIntoView, stripIdVersion } from '../utils';
+import { ArchiveTree, Book, Page, State } from '../types';
+import { scrollTocSectionIntoView } from '../utils/domUtils';
+import { stripIdVersion } from '../utils/idUtils';
+import {
+  bookBannerDesktopHeight,
+  bookBannerMobileHeight,
+  sidebarDesktopWidth,
+  sidebarMobileWidth,
+  sidebarTransitionTime,
+  toolbarDesktopHeight,
+  toolbarMobileHeight
+} from './constants';
 import ContentLink from './ContentLink';
+import SidebarControl from './SidebarControl';
+import { styleWhenSidebarClosed } from './utils/sidebar';
 
-export const sidebarWidth = 33.5;
-const sidebarTransitionTime = 300;
 const sidebarPadding = 1;
 
-// TODO - magic numbers to be replaced in `top`, `height` when ToC gets real styling in openstax/unified#176
-const SidebarBody = styled.div<{isOpen: boolean}>`
-  top: 23rem;
-  height: calc(100vh - 23rem);
+const sidebarClosedStyle = css`
+  overflow-y: hidden;
+  transform: translateX(-${sidebarDesktopWidth}rem);
```

Comment:
Are these `translateX` here for the animations or to get around having parent elements know about the ToC state? (or both?)